docs: chap1_고현주 md 파일 추가

theeeeeng opened the pull request:
이제 집으로...... 😢

goodockei committed:
docs: chap1_고현주 md 파일 추가

appear commented on chapter_1/chap1_고현주.md:
```diff
@@ -0,0 +1,221 @@
+
+> MIME 타입
+
+웹 서버는 모든 HTTP 객체 데이터에 MIME 타입을 붙입니다. 브라우저는 서버로부터 객체를 돌려받을 때, 다룰 수 있는 개체인지 MIME 타입을 통해 확인할 수 있는데, 대부분의 브라우저는 보편적인 타입 수백 가지를 다룰 수 있다고 합니다.
```
```
> MIME 타입

> 웹 서버는 모든 HTTP 객체 데이터에 MIME 타입을 붙입니다. 브라우저는 서버로부터 객체를 돌려받을 때, 다룰 수 있는 개체인지 MIME 타입을해 확인할 수 있는데, 대부분의 브라우저는 보편적인 타입 수백 가지를 다룰 수 있다고 합니다.
```

요런 느낌으로 써주셔야 의도한것처럼 나올 것 같아요 ..!?

theeeeeng replied:
>를 항목?표시용으로 적어서 저렇게 적었는데, 아래 적어주신 내용을 보니 MIME 타입에 >를 붙인것 자체를 변경해야겠네요 ㅠㅠ 다음 글부터는 조언해주신대로 항목정리를 해보겠습니다 😃

appear commented on chapter_1/chap1_고현주.md:
```diff
@@ -0,0 +1,221 @@
+
+예를 들어, 암호화된 SSL 트래픽을 HTTP 커넥션으로 전송함으로써 웹 트래픽만 허용하는 사내 방화벽을 통과시키는 터널이 있습니다. HTTP/SSL 터널은 HTTP 요청을 받아들여 목적지의 주소와 포트번호로 커넥션을 맺어서 이후부터는 암호화된  SSL 트래픽을 HTTP 채널을 통해 목적지 서버로 전송할 수 있게 됩니다.
+
+> 에이전트
```
`>` 태그가 많이 보이는데 의도하신 스타일인것 같은데 (갠취이긴하오나) `>` 태그의 경우 부가 설명이나 소소한 여담 같은것을 적어줄때 좋은 것 같아요 적어주신것들은 sub title 같은 느낌이라서 좀 더 제목스럽게 보이기 위해서는 다른 태그를 이용하는게 좋지 않을까라는 ... 생각이듭니다 :)

theeeeeng replied:
아하...!! 감사합니다!! 깔끔한 글을 쓰는게 아직 너무 어려워서 이런 피드백 대환영입니다!!!! :clap

appear reviewed (approved):
👍

feel5ny reviewed (commented):
README.md 파일에 링크 추가해주셔야할꺼같습니다! :)

feel5ny reviewed (approved):
1장을 잘 정리해주신거같습니다.
실제 크롬에서 나온는 메세지를 캡쳐해주신 부분 덕분에 이해가 더 잘 된거같습니다.
저도 석진님과 같은 의견으로 blockquote보다는 heading을 사용하는게 어떨까 합니다.(물론 개인 스타일이라 ~)
고생하셨습니다 !

feel5ny commented on chapter_1/chap1_고현주.md:
```diff
@@ -0,0 +1,221 @@
+
+> MIME 타입
+
+웹 서버는 모든 HTTP 객체 데이터에 MIME 타입을 붙입니다. 브라우저는 서버로부터 객체를 돌려받을 때, 다룰 수 있는 개체인지 MIME 타입을 통해 확인할 수 있는데, 대부분의 브라우저는 보편적인 타입 수백 가지를 다룰 수 있다고 합니다.
```
👀
개체인지 => 객체인지

theeeeeng replied:
:scre

theeeeeng committed:
오타수정

brainbackdoor commented on chapter_1/chap1_고현주.md:
```diff
@@ -0,0 +1,221 @@
+
+위에서 이야기한 이미지, HTML 문서, 텍스트, 동영상 등의 리소스들은 웹 서버에 있습니다. 웹 서버(이하 HTTP 서버)는 HTTP 클라이언트가 보낸 데이터를 저장하고, HTTP 클라이언트가 요청한 데이터를 제공해줍니다.
+
+웹 브라우저(HTTP 클라이언트)가 서버에 HTTP 객체를 요청하면 서버가 보내준 리소스를 웹 브라우저가 처리해줍니다. 예를 들어, 크롬에서 [http://theeeeeng.github.com](https://theeeeeng.github.com) 링크로 접근하면, 크롬은  [http://theeeeeng.github.com](https://theeeeeng.githumb.com) 서버로 HTTP 요청(여기서는 index.html 요청)을 보냅니다. 요청받은 서버는 요청받은 객체를 찾고, 성공했다면 요청받은 객체의 타입, 길이 등의 정보와 함께 HTTP 응답에 실어서 클라이언트로 전송합니다.
```
```suggestion
웹 브라우저(HTTP 클라이언트)가 서버에 HTTP 객체를 요청하면 서버가 보내준 리소스를 웹 브라우저가 처리해줍니다. 예를 들어, 크롬에서 [http://theeeeeng.github.com](https://theeeeeng.github.com) 링크로 접근하면, 크롬은  [http://theeeeeng.github.com](https://theeeeeng.githumb.com) 서버로 HTTP 요청을 보냅니다. 요청받은 서버는 요청받은 객체를 찾고, 성공했다면 요청받은 객체의 타입, 길이 등의 정보와 함께 HTTP 응답에 실어서 클라이언트로 전송합니다.
```

Browser가 index.html resource를 요청하는 것이 아니라 Server에 설정되어 있는 default file을 리턴하는 것으로 알고 있어요 

https://stackoverflow.com/questions/22121056/how-to-make-a-website-run-the-index-html-file

theeeeeng replied:
아하, default file 이라는 의미로 **여기서는 index.html** 이라는 표현을 썼는데 오히려 더 헷갈리는 표현이 되었네요 ㅠㅠ 피드백 감사합니다!!!!!!! 👍

brainbackdoor commented on chapter_1/chap1_고현주.md:
```diff
@@ -0,0 +1,221 @@
+
+HTTP 요청/응답 메시지의 형식은 비슷합니다.
+
+요청 메세지는 `시작줄/헤더`로 구성되고, 응답 메세지는 `시작줄/헤더/본문`으로 구성되어 있습니다.
```
```suggestion
요청/응답 메세지는 `시작줄/헤더/본문`으로 구성되어 있습니다.
```
>https://tools.ietf.org/html/rfc2616#page-32

`The message-body (if any) of an HTTP message is used to carry the
   entity-body associated with the request or response.`


12 page 본문에도 `요청의 본문은 웹 서버로 데이터를 실어 보내며, 응답의 본문은 클라이언트로 데이터를 반환한다.`고 나와 있네요.

theeeeeng replied:
수정 감사합니다 :)

brainbackdoor commented on chapter_1/chap1_고현주.md:
```diff
@@ -0,0 +1,221 @@
+
+- 응답 시작줄
+
+    ![](Untitled-9a56ba46-0225-4903-ace2-8f5662ed4345.png)
```
👍

brainbackdoor commented on chapter_1/chap1_고현주.md:
```diff
@@ -0,0 +1,221 @@
+
+HTTP 클라이언트가 서버에 메세지를 전송하기 전에, IP 와 포트번호를 사용해 클라이언트와 서버 사이에 TCP/IP 커넥션을 맺어야합니다.
+
+[http://0.0.0.0:8080/index.html](http://0.0.0.0:8080/index.html) URL 에서 **IP는 0.0.0.0** 이고, **포트번호는 80**입니다.
```
0.0.0.0은 전체 대역대를 의미하는 IP라 예로 들기에는 어색한거 같아요.

https://en.wikipedia.org/wiki/0.0.0.0

brainbackdoor reviewed (approved):
직접 캡쳐까지 하셨네요! 👍 
덕분에 재밌게 잘 읽었습니다. ㅎㅎ 
몇가지 제가 알고 있던 것과 다른 부분이 있어 피드백 남겼습니다. 확인해보시고 이상 있으시면 공유 부탁드려요~ 😀

yurimkoo reviewed (approved):
책에 나온 예시가 아닌 실제 예시를 캡쳐해서 첨부해주신 점이 좋았어요! 저도 다른 분들처럼 인용문자 보다는 서브 타이틀임을 명확하게 알려줄 수 있는 강조 표현을 사용하는게 좋을 것 같아요. :)

junjangsee commented:
잘 읽었습니다! 저와 비슷한 마크다운 활용을 사용하고 계신것 같네요 ㅎㅎ 작성하신 것처럼 앞으로 저도 실제로 클라이언트 요청을 해보면서 기록을 남기면 더 머리에 남을 것 같아요 !

theeeeeng committed:
Update chapter_1/chap1_고현주.md

https://github.com/bookcrush/httpPerfectGuide/pull/5#discussion_r310845693

Co-Authored-By: DongGyu Lee <brainbackdoor@gmail.com>

Jongjineee commented:
그냥 요약만 하신게 아니군요! 내용 정리가 잘 되어 있어 이해하기 편했습니다! 😃

Jongjineee reviewed (approved):
그냥 요약만 하신게 아니군요! 내용 정리가 잘 되어 있어 이해하기 편했습니다! 😃

theeeeeng committed:
내용 수정

요청 메세지는 `시작줄/헤더`로 구성되고, 응답 메세지는 `시작줄/헤더/본문`으로 구성되어 있습니다. -> 요청/응답 메세지는 `시작줄/헤더/본문`으로 구성되어 있습니다.

bebiangel commented:
이해하기 편하도록 이미지 첨부한게 좋네요!
그리고 제목과 내용을 적절하게 분리해서 가독성도 좋아서 읽기 편했습니다.

Jeewhan commented:
마크다운을 잘 활용하시고, 중간 중간에 이미지도 알맞게 첨부되어있어서 정말 좋았습니다!!
